t/t5516-fetch-push.sh

```diff
@@ -1712,4 +1712,23 @@ test_expect_success 'updateInstead with push-to-checkout hook' '
 	)
 '
 
+test_expect_success 'denyCurrentBranch and worktrees' '
+    git worktree add new-wt &&
+	git clone . cloned &&
+	test_commit -C cloned first &&
+	test_config receive.denyCurrentBranch refuse &&
+	test_must_fail git -C cloned push origin HEAD:new-wt &&
+	test_config receive.denyCurrentBranch updateInstead &&
+	git -C cloned push origin HEAD:new-wt &&
+	test_must_fail git -C cloned push --delete origin new-wt
+'
+
+test_expect_failure 'denyCurrentBranch and unborn branch' '
+	git init repo &&
+	git clone repo cloned-repo &&
+	test_commit -C cloned-repo first &&
+	test_config receive.denyCurrentBranch updateInstead &&
```

Comment:
If you take pains to create two separate repositories for this test case, you will also need to take pains to set the config variable in the intended repository (it is currently not set in the correct repository).

Reply:
(As I said, I would not even bother with the `cloned-repo`...)